scripts/feed.js

```diff
@@ -32,39 +38,53 @@ function Feed(feed_urls)
     this.get_entries();
   }
 
-  this.get_entries = async function()
+  this.get_entries = function()
   {
     var entries = [];
-
-    var portal_summary = "";
-
-    for(id in this.archives){
-      var archive = this.archives[id];
-      var portal_data;
-      try {
-        portal_data = await archive.readFile('portal.json');
-      } catch (e) {
-        console.warn(`Unable to fetch, this feed appears to be offline: ${archive.url}`);
-      }
-      var portal = JSON.parse(portal_data);
-      this.portals[portal.name] = archive.url;
-
-      for(entry_id in portal.feed){
-        var entry_data = portal.feed[entry_id];
-        entry_data.portal = portal.name;
-        entry_data.dat = archive.url;
-        entry_data.id = entry_id;
-        entries.push(new Entry(entry_data))
-      }
-    }
-
-    // Sort
-
-    var sorted_entries = entries.sort(function(a,b){
-       return a.timestamp < b.timestamp ? -1 : 1;
+    var online_ports_count = 0;
+
+    var archive_promises = this.archives.map((archive) => (
+      this.get_feed(archive)
+        .then((feed_entries) => {
+          online_ports_count += 1;
+          r.portal.port_status_el.innerHTML = (online_ports_count - 1)+" online";
+
+          entries = entries.concat(feed_entries);
+
+          // Sort
+          var sorted_entries = entries.sort(function(a,b){
+            return a.timestamp < b.timestamp ? -1 : 1;
+          });
+
+          this.refresh(sorted_entries.reverse());
```

Comment:
It sorts and re-renders every time a port is loaded, perhaps not the most efficient thing but it should be ok for now.